tests/nodetree/test_node_functional.py

```diff
@@ -0,0 +1,383 @@
+from __future__ import annotations
+
+import pickle
+from enum import Enum
+from io import BytesIO
+
+import pytest
+from labone.core.value import AnnotatedValue
+from labone.errors import LabOneError
+from labone.nodetree.enum import _get_enum
+
+from tests.mock_server_for_testing import get_mocked_node, get_unittest_mocked_node
+
+
+@pytest.mark.asyncio()
+async def test_node_dot_subpathing():
+    node = await get_unittest_mocked_node({"/a/b/c/d": {}})
+
+    assert node.a.b.c.d.path == "/a/b/c/d"
+
+
+@pytest.mark.asyncio()
+async def test_node_bracket_subpathing():
+    node = await get_unittest_mocked_node({"/a/b/c/d": {}})
+
+    assert node["a"]["b"]["c"]["d"].path == "/a/b/c/d"
+
+
+@pytest.mark.asyncio()
+async def test_node_bracket_path_subpathing():
+    node = await get_unittest_mocked_node({"/a/b/c/d": {}})
+
+    assert node["a/b/c/d"].path == "/a/b/c/d"
+
+
+@pytest.mark.asyncio()
+async def test_node_bracket_number_subpathing():
+    node = await get_unittest_mocked_node({"/0": {}})
+    assert node[0].path == "/0"
+
+
+@pytest.mark.asyncio()
+async def test_node_bracket_wildcard_subpathing():
+    node = await get_unittest_mocked_node({"/a/b": {}})
+    assert node["*"].b.path == "/*/b"
+
+
+@pytest.mark.asyncio()
+async def test_subpathing_invalid_path_raises():
+    node = await get_unittest_mocked_node({"/a": {}})
+    with pytest.raises(Exception):  # noqa: B017
+        node.b  # noqa: B018
+
+
+@pytest.mark.asyncio()
+async def test_subpathing_too_deep_path_raises():
+    node = await get_unittest_mocked_node({})
+    with pytest.raises(Exception):  # noqa: B017
+        node.b  # noqa: B018
```

Comment:
suggestion: This could also raise because the nodetree is empty. 
```suggestion
    node = await get_unittest_mocked_node({"/a": {}})
    with pytest.raises(Exception):  # noqa: B017
        node.a.b  # noqa: B018
```